modules/pipelines/worker.py

```diff
@@ -230,86 +244,108 @@ def worker(
         if not is_valid:
             raise ValueError(f"Invalid parameters: {error_message}")
-        
+
         # Prepare parameters
         job_params = pipeline.prepare_parameters(job_params)
-        
+
         if not high_vram:
             # Unload everything *except* the potentially active transformer
             unload_complete_models(text_encoder, text_encoder_2, image_encoder, vae)
-            if studio_module.current_generator is not None and studio_module.current_generator.transformer is not None:
-                offload_model_from_device_for_memory_preservation(studio_module.current_generator.transformer, target_device=gpu, preserved_memory_gb=8)
-
+            if current_generator is not None and current_generator.transformer is not None:
+                offload_model_from_device_for_memory_preservation(
+                    current_generator.transformer, target_device=gpu, preserved_memory_gb=settings.get("gpu_memory_preservation", 8.0))
 
         # --- Model Loading / Switching ---
         print(f"Worker starting for model type: {model_type}")
-        print(f"Worker: Before model assignment, studio_module.current_generator is {type(studio_module.current_generator)}, id: {id(studio_module.current_generator)}")
-        
+        print(
+            f"Worker: Before model assignment, current_generator is {type(current_generator)}, id: {id(current_generator)}")
+
         # Create the appropriate model generator
-        new_generator = create_model_generator(
-            model_type,
-            text_encoder=text_encoder,
-            text_encoder_2=text_encoder_2,
-            tokenizer=tokenizer,
-            tokenizer_2=tokenizer_2,
-            vae=vae,
-            image_encoder=image_encoder,
-            feature_extractor=feature_extractor,
-            high_vram=high_vram,
-            prompt_embedding_cache=prompt_embedding_cache,
-            offline=args.offline,
-            settings=settings
+        if studio_manager.is_reload_required(
+            model_name=model_type,
+            selected_loras=selected_loras,
+            lora_values=lora_values,
+            lora_loaded_names=lora_loaded_names
+        ):
+            new_generator = create_model_generator(
+                model_type,
+                text_encoder=text_encoder,
+                text_encoder_2=text_encoder_2,
+                tokenizer=tokenizer,
+                tokenizer_2=tokenizer_2,
+                vae=vae,
+                image_encoder=image_encoder,
+                feature_extractor=feature_extractor,
+                high_vram=high_vram,
+                prompt_embedding_cache=prompt_embedding_cache,
+                offline=args.offline,
+                settings=settings
+            )
+
+            studio_manager.current_generator = current_generator = new_generator
+            # Load the transformer model
+            current_generator.load_model()
```

Comment:
It's not clear why we load here without moving the transformer to the gpu, but on 296 a pre-existing transformer does go to the gpu.